bld/CLMBuildNamelist.pm

```diff
@@ -2316,6 +2316,17 @@ sub setup_logic_do_transient_pfts {
    # generated
    my $cannot_be_true = "";
 
+   my $n_dom_pfts = $nl->get_value( 'n_dom_pfts' );
+   my $n_dom_landunits = $nl->get_value( 'n_dom_landunits' );
+   my $toosmall_soil = $nl->get_value( 'toosmall_soil' );
+   my $toosmall_crop = $nl->get_value( 'toosmall_crop' );
+   my $toosmall_glacier = $nl->get_value( 'toosmall_glacier' );
+   my $toosmall_lake = $nl->get_value( 'toosmall_lake' );
+   my $toosmall_wetland = $nl->get_value( 'toosmall_wetland' );
+   my $toosmall_urb_tbd = $nl->get_value( 'toosmall_urb_tbd' );
```

Comment:
@slevisconsulting a question that comes to my mind for urban is if people would really want to break it out into the three different landunit types. This may be the right way to do it for flexibility, and just the natural way that the code works. But, I'd like to check with people about what might be desired. @barlage and @negin513 can you maybe comment here? I wonder if the desired behavior might more be toward turning urban off completely, or off if the total urban is more than some total amount? Would there be a use for tweaking the three different types differently? I'm not sure, so it's worth talking to some people who would be using this feature.

Reply:
@ekluzek it may be that I just don't understand the nomenclature, but I'm not quite sure what you are asking.

Reply:
@barlage Urban is divided into three different types: tall building district, high density, and medium density. What I'm thinking is that you probably aren't going to want to treat the three types differently. And more than that, since you don't care about the different types of urban, you probably really want to control it based on the total amount of urban in the grid-cell -- regardless of type. Having it broken up into the three types doesn't allow you to have control over total urban in the grid cell. So I'm thinking that having the namelist control item be toosmall_urban (rather than toosmall_urb_(tbd,hd,md)) would be more useful for users. And toosmall_urban is probably more obvious as to it's meaning.

Reply:
@ekluzek OK, I get it. I do like the added flexibility to say e.g., if the total urban (tbd+hd+md) is below some threshold, don't use it; if it is above, do something else (like use dominant of tbd, hd, md). I can tell you that currently in WRF, in "default" mode, only the dominant of all types (including the tbd, hd, md equivalent) is used to define the grid (to complicate things, if it is one of these "urban" types, the non-urban fraction will be a vegetation class...). This dominant approach can underestimate urban cover and was one of the main motivations for adding a "mosaic" option to the Noah LSM in WRF. When using this option, the n most dominant vegetation classes are used, still regardless of similar typing (like total urban, total forests, etc.).

Reply:
@barlage let me say it a bit differently:
- The way I wrote the code, you can set 3 "toosmall_urb" namelist parameters to decide whether one urban type is more important to you than another. So you may keep the tbd (tall building districts) and remove the hd and md, or vice versa, or keep no urban, or keep all urban, or any combination you choose.
- The suggested alternative simplifies this to one "toosmall_urban" namelist parameter, but you end up with all three urban landunits or lose all three.

I am happy to go with the alternative if you prefer it. I just wanted to make sure.

Reply:
Just to be clear, I will not change the implementation until I get the go-ahead that people prefer the alternative.

Reply:
@ekluzek , @negin513 , @barlage , and I decided that I will modify to the following:

Order of new subroutine calls at the end of subroutine surfrd_get_data:
1) Newest call will happen first: collapse_urban = .true. or .false.  If true, this will put the three urban landunit areas into the dominant urban landunit.
2) Recently added call using n_dom_landunits will not change
3) Recently added call using too_small_* will be modified so that a single too_small_urban value (namelist controlled) will apply to each urban landunit that's present.

Reply:
From above list...
Step 3: complete in b3dcad9
Step 2: no change
Step 1: I will work on next

Reply:
Step 1 complete in 99645bb